src/hotspot/share/oops/markOop.hpp

```diff
@@ -131,6 +131,9 @@ class markOopDesc: public oopDesc {
          biased_lock_mask         = right_n_bits(lock_bits + biased_lock_bits),
          biased_lock_mask_in_place= biased_lock_mask << lock_shift,
          biased_lock_bit_in_place = 1 << biased_lock_shift,
+         // self forward bit has the same position of biased lock bit
```

Comment:
what's the relation between using biased locking and using self-forward bit? are these two features mutually exclusive

Reply:
self-forwarding is used for “forwarded” state with lock bits “11” while biased locking bit is not used in marked/forwarded state.